sunpy/map/mapbase.py

```diff
@@ -2819,7 +2824,38 @@ def plot(self, *, annotate=True, axes=None, title=True, autoalign=False,
         else:
             data = np.ma.array(np.asarray(self.data), mask=self.mask)
 
-        if autoalign == 'pcolormesh':
+        if autoalign == 'image':
+            ny, nx = self.data.shape
+            pixel_perimeter = grid_perimeter(nx, ny) - 0.5
+
+            transform = axes.get_transform(self.wcs) - axes.transData
+            data_perimeter = transform.transform(pixel_perimeter)
+            data_corners = data_perimeter[[0, nx, nx + ny, 2*nx + ny], :]
```

Comment:
Probably better to move this below the RuntimeError block?

Reply:
Sure